Proteomics/RetentionTimePrediction/SSRCalc3.cs

```diff
@@ -0,0 +1,1486 @@
+            if (sze < 10)                         // short peptides use short peptide retention weights
+            {
+                tsum3 =
+                   AAPARAMS[seq[0]].RC1S +        // Sum weights for 1st
+                   AAPARAMS[seq[1]].RC2S +        // second,
+                   AAPARAMS[seq[sze - 1]].RCNS +    // ultimate
+                   AAPARAMS[seq[sze - 2]].RCN2S;    // and penultimate aa
+
+                for (i = 2; i < sze - 2; i++)            // add weights for aa's in the middle
+                {
+                    tsum3 += AAPARAMS[seq[i]].RCS;
+                }
+            }
+            else                                  // longer peptides use regular retention weights
+            {
+                tsum3 =
+                   AAPARAMS[seq[0]].RC1 +         // Sum weights for 1st
+                   AAPARAMS[seq[1]].RC2 +         // second,
+                   AAPARAMS[seq[sze - 1]].RCN +     // ultimate
+                   AAPARAMS[seq[sze - 2]].RCN2;     // and penultimate aa
+
+                for (i = 2; i < sze - 2; i++)            // add weights for aa's in the middle
+                {
+                    tsum3 += AAPARAMS[seq[i]].RC;
+                }
+            }
+            //_log.debug("Core = "+tsum3);
+
+            // 1- smallness - adjust based on tsum score of peptides shorter than 20 aa's.
+            tsum3 += smallness(sze, tsum3);
+            //_log.debug("smallness = "+tsum3);
+            // 2- undigested parts
+            tsum3 -= undigested(seq);
+            //_log.debug("undigested = "+tsum3);
+            // 3- clusterness # NB:weighting of v1 is now done in subrtn.
+            tsum3 -= clusterness(seq);
+            //_log.debug("clusterness = "+tsum3);
+            // 4- proline fix
+            tsum3 -= proline(seq);
+            //_log.debug("proline = "+tsum3);
+            // 5- length scaling correction
+            tsum3 *= length_scale(sze);
+            //_log.debug("length_scale = "+tsum3);
+            // 6- total sum correction
+            if (tsum3 >= 20 && tsum3 < 30) tsum3 -= ((tsum3 - 18) * SUMSCALE1);
+            if (tsum3 >= 30 && tsum3 < 40) tsum3 -= ((tsum3 - 18) * SUMSCALE2);
+            if (tsum3 >= 40 && tsum3 < 50) tsum3 -= ((tsum3 - 18) * SUMSCALE3);
+            if (tsum3 >= 50) tsum3 -= ((tsum3 - 18) * SUMSCALE4);
+            //_log.debug("total sum = "+tsum3);
+            // 7- isoelectric change
+            tsum3 += newiso(seq, tsum3);
+            //_log.debug("isoelectric = "+tsum3);
+            // 8- helicity corrections  #NB: HELIX#SCALE-ing is now done in subrtn.
+            tsum3 += helicity1(seq);
+            //_log.debug("helicity1 = "+tsum3);
+            tsum3 += helicity2(seq);
+            //_log.debug("helicity2 = "+tsum3);
+            tsum3 += helectric(seq);
+            //_log.debug("helectric = "+tsum3);
+            return tsum3;
+        }
+
+        private double smallness(int sqlen, double tsum)
```

Comment:
Could you make the method names all PascalCase for C# style consistency?

Reply:
done